stl/inc/xstring

```diff
@@ -1285,6 +1289,25 @@ public:
     constexpr basic_string_view(_It _First, _Se _Last) noexcept(noexcept(_Last - _First)) // strengthened
         : _Mydata(_STD to_address(_First)), _Mysize(static_cast<size_type>(_Last - _First)) {}
     // clang-format on
+
+#if _HAS_CXX23
+    // clang-format off
+    template <_RANGES contiguous_range _Range>
+        requires _RANGES sized_range<_Range>
+            && same_as<_RANGES range_value_t<_Range>, _Elem>
+            && (!is_convertible_v<_Range, const _Elem*>)
+            && (!requires(remove_cvref_t<_Range>& _Rng) {
+                _Rng.operator _STD basic_string_view<_Elem, _Traits>();
+            })
+            && (!requires {
+                // per editorial or LWG issue not yet filed as of 2021-06-10
+                typename remove_reference_t<_Range>::traits_type;
+            } || same_as<typename remove_reference_t<_Range>::traits_type, _Traits>)
+    constexpr basic_string_view(_Range&& _Rng) noexcept(
+        noexcept(_RANGES data(_Rng)) && noexcept(_RANGES size(_Rng))) // strengthened
+        : _Mydata(_RANGES data(_Rng)), _Mysize(_RANGES size(_Rng)) {}

```

Comment:
We need to `static_cast` the result of `ranges::size` to `size_t` here since (1) it might be signed, and we don't want the compiler to complain about the conversion which we know is valid, and (2) it might be an integer-class type which is explicitly-but-not-implicitly convertible to `size_t`.

I've submitted an LWG issue to add this `static_cast` to the pertinent _Effects_ element in the working draft.

Reply:
> (2) it might be an integer-class type which is explicitly-but-not-implicitly convertible to `size_t`.

This is direct-initialization so it is already capable of using explicit conversion functions.

Reply:
> > (2) it might be an integer-class type which is explicitly-but-not-implicitly convertible to `size_t`.
> 
> This is direct-initialization so it is already capable of using explicit conversion functions.

Bah - thanks, Tim. That just leaves reason (1), and I need to retract my LWG issue submission.